### PR TITLE
Support latest version of build, upgrade analyzer dev

### PR DESCRIPTION
**Pull request by kevmoo**
Also stop using most deprecated analyzer APIs

### CODE REVIEW

File: lib/src/utils.dart [hunk 1]
```diff
@@ -8,19 +8,19 @@ import 'dart:async';
...
 import 'package:analyzer/source/package_map_resolver.dart';
 import 'package:analyzer/source/pub_package_map_provider.dart';
-import 'package:analyzer/src/generated/ast.dart';
-import 'package:analyzer/src/generated/element.dart';
 import 'package:analyzer/src/generated/engine.dart';
 import 'package:analyzer/src/generated/java_io.dart';
-import 'package:analyzer/src/generated/sdk_io.dart' show DirectoryBasedDartSdk;
 import 'package:analyzer/src/generated/sdk.dart' show DartSdk;
-import 'package:analyzer/src/generated/source_io.dart';
+import 'package:analyzer/src/generated/sdk_io.dart' show DirectoryBasedDartSdk;
```
Comment (jakemac53) [inline lib/src/utils.dart]: This (and the entire `getAnalysisContextForProjectPath` method) shouldn't actually be necessary any more? I must have missed deleting that when I switched to pkg/build.

Comment (jakemac53) [inline lib/src/utils.dart]: Actually now that I think about it I think particle is using this though... and it looks like this package is using it in tests.

**Comment by jakemac53** [comment]
lgtm, may want to move `getAnalysisContextForProjectPath` to the test folder or something, but since it is being relied on externally maybe not.

File: pubspec.yaml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 author: Dart Team <misc@dartlang.org>
 description: Automatic sourcecode generation for Dart
 homepage: https://github.com/dart-lang/source_gen
 environment:
   sdk: '>=1.12.0 <2.0.0'
 dependencies:
-  analyzer: '^0.27.3-alpha.1'
-  build: ^0.2.1
+  analyzer: '^0.27.3'
+  build: '>=0.2.1 <0.4.0'
```
Comment (jakemac53) [inline pubspec.yaml]: Technically the tests wouldn't pass in 0.2.x, but I think thats ok. It does actually support that version, its just the enum values which appear in the test that would be broken.

Comment (kevmoo) [inline pubspec.yaml]: Exactly 😄

File: lib/src/utils.dart (hunk 1 above)
Comment (kevmoo) [inline lib/src/utils.dart]: ...so leave it as is?

**Pull request merged**